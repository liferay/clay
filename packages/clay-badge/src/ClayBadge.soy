{namespace ClayBadge}

/**
 * This renders a badge.
 */
{template .render}
	{@param label: string}
	{@param? elementClasses: string}
	{@param? id: string}
	{@param? style: string}

	{let $attributes kind="attributes"}
		class="badge
			{sp}badge-{$style ?: 'primary'}

			{if $elementClasses}
				{sp}{$elementClasses}
			{/if}
		"

		{if $id}
			id="{$id}"
		{/if}
	{/let}

	<span {$attributes}>
		<span class="badge-item badge-item-expand">{$label}</span>
	</span>
{/template}
