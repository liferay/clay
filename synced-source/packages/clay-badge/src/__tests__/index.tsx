/**
 * SPDX-FileCopyrightText: Â© 2019 Liferay, Inc. <https://liferay.com>
 * SPDX-License-Identifier: BSD-3-Clause
 */

import ClayBadge from '..';
import React from 'react';
import TestRenderer from 'react-test-renderer';

describe('ClayBadge', () => {
	it('renders', () => {
		const testRenderer = TestRenderer.create(
			<ClayBadge displayType="primary" label="4" />
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('applies custom className', () => {
		const testRenderer = TestRenderer.create(
			<ClayBadge className="bar foo" displayType="primary" label="4" />
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('renders translucent', () => {
		const testRenderer = TestRenderer.create(
			<ClayBadge displayType="info" label="Beta" translucent />
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('renders dark and translucent', () => {
		const testRenderer = TestRenderer.create(
			<ClayBadge dark displayType="info" label="Beta" translucent />
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('renders displayType="beta" as info and translucent', () => {
		const testRenderer = TestRenderer.create(
			<ClayBadge displayType="beta" label="Beta" />
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('renders displayType="beta-dark" as dark, info and translucent', () => {
		const testRenderer = TestRenderer.create(
			<ClayBadge displayType="beta-dark" label="Beta" />
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});
});
