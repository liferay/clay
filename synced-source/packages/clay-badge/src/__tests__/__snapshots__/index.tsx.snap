// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayBadge applies custom className 1`] = `
<span
  className="badge badge-primary bar foo"
>
  <span
    className="badge-item badge-item-expand"
  >
    4
  </span>
</span>
`;

exports[`ClayBadge renders 1`] = `
<span
  className="badge badge-primary"
>
  <span
    className="badge-item badge-item-expand"
  >
    4
  </span>
</span>
`;

exports[`ClayBadge renders dark and translucent 1`] = `
<span
  className="badge badge-info badge-translucent clay-dark"
>
  <span
    className="badge-item badge-item-expand"
  >
    Beta
  </span>
</span>
`;

exports[`ClayBadge renders displayType="beta" as info and translucent 1`] = `
<span
  className="badge badge-info badge-translucent"
>
  <span
    className="badge-item badge-item-expand"
  >
    Beta
  </span>
</span>
`;

exports[`ClayBadge renders displayType="beta-dark" as dark, info and translucent 1`] = `
<span
  className="badge badge-info badge-translucent clay-dark"
>
  <span
    className="badge-item badge-item-expand"
  >
    Beta
  </span>
</span>
`;

exports[`ClayBadge renders translucent 1`] = `
<span
  className="badge badge-info badge-translucent"
>
  <span
    className="badge-item badge-item-expand"
  >
    Beta
  </span>
</span>
`;
